keras/callbacks.py

```diff
@@ -779,11 +796,35 @@ def on_epoch_end(self, epoch, logs=None):
                     self.writer.add_summary(summary_str, epoch)
                     i += self.batch_size
 
-        if self.embeddings_freq and self.embeddings_ckpt_path:
+        if self.embeddings_freq and self.embeddings_data is not None:
             if epoch % self.embeddings_freq == 0:
-                self.saver.save(self.sess,
-                                self.embeddings_ckpt_path,
-                                epoch)
+
+                embeddings_data = self.embeddings_data
+                n_samples = embeddings_data.shape[0]
+
+                i = 0
+                while i < n_samples:
+                    if type(self.model.input) == list:
+                        feed_dict = {model_input: self.embeddings_data
+                                     for model_input in list(self.model.input)}
+                    else:
+                        feed_dict = {self.model.input: self.embeddings_data}
+
+                    step = min(self.batch_size, n_samples - i)
+                    if self.model.uses_learning_phase:
+                        feed_dict[K.learning_phase()] = False
+                        # do not slice the learning phase
+                        batch_val = [x[i:i + step] for x in embeddings_data[:-1]]
+                        batch_val.append(embeddings_data[-1])
```

Comment:
I don't think `embeddings_data` should have to be passed with the learning phase appended to the end.